posthog/src/test/java/com/posthog/internal/PostHogApiTest.kt

```diff
@@ -130,6 +138,35 @@ internal class PostHogApiTest {
         assertEquals("application/json; charset=utf-8", request.headers["Content-Type"])
     }
 
+    @Test
+    fun `client uses configured proxy for requests`() {
+        val file = File("src/test/resources/json/basic-remote-config.json")
+        val responseApi = file.readText()
+
+        val http =
+            mockHttp(
+                response =
+                    MockResponse()
+                        .setBody(responseApi),
+            )
+
+        val hostname = "localhost"
+        val port = 8080
+        val proxyAddress = InetSocketAddress(hostname, port)
+        val proxy = Proxy(Proxy.Type.HTTP, proxyAddress)
+
+        val url = http.url("/")
+
+        val sut = getSut(host = url.toString(), proxy)
+        val exception = try {
+            sut.remoteConfig()
+            null
+        } catch (e: Exception) {
+            e
+        }
+        assertNotNull(exception)
```

Comment:
I dont think this is asserting anything, we need to make sure that the HTTP request is being made using the `hostname` and `port` you set